### PR TITLE
docs: create dependency dashboard showcase page

### LINKED ISSUE #11178: Add Dependency Dashboard showcase page
**What would you like Renovate to be able to do?**

A user [commented](
https://github.com/renovatebot/renovate/issues/11134#issuecomment-894242852) about our Dependency Dashboard:

>Awesome feature. It should have a better exposure in the documentation maybe with a dedicated section in getting started.

I also think the dashboard is a great feature, and I agree that we should promote it better, and increase the discoverability of the feature.

We need this page to be there, before we can make "Dependency Dashboard enabled" the default behavior in `v26`.

**Did you already have any implementation ideas?**

It would probably go well under the Getting Started section of the sidebar, with a new entry "Dependency Dashboard".

I could write sections on:

- Supported platforms
- How to enable the dashboard
- Use-cases:
  - Dependency Dashboard Approval workflow
  - Visibility into rejected/deferred updates
  - Other things that I'm not thinking of right now? Open for suggestions! 😉 
  
Let me know if this is something you want me to work on. And if you have ideas on how to make it better, please share

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Create dependency Dashboard showcase page

## Context:

Fixes #11178.

## Todo's:

- [x] Get feedback on draft PR
- [x] Resolve all `TODO:` in the current PR
- [x] Discuss whether we want to add screenshots
- [x] Add the new page to correct place in sidebar in the Renovate Docs repository, edit file: https://github.com/renovatebot/renovatebot.github.io/blob/build/mkdocs.yml

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by HonkingGoose** [comment]
Do we want to have screenshots of the dashboard?

**Benefits:**

- Pictures help to liven up the page so it's not just a wall of text
- Pictures are a good way to show why you want to have the dashboard
- Helps market the feature to new prospecting users who are reading the docs to find out if Renovate is a good fit for them

**Drawbacks:**

- Pictures will need to be updated to match current design (example when we implement issue `#10924`)
- Increases size of repository a bit
- Requires maintainer time to update pictures

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [inline docs/usage/dependency-dashboard.md:22]: I noticed we have a preset to enable the dashboard, but we don't have a preset to turn it off. So maybe we want to rejig things so that:

- `:enableDependencyDashboard` -> turns on dashboard
- `:disableDependencyDashboard` -> turns off dashboard (if it's enabled by default like in `v26`)

Don't know if this is something we want/like?

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [inline docs/usage/dependency-dashboard.md:46]: https://docs.renovatebot.com/configuration-options/#dependencydashboardapproval says to set `dependencyDashboardApproval` to `true`, but we also have the `:dependencyDashboard` preset, which is not mentioned in the link.

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [inline docs/usage/dependency-dashboard.md:50]: If we decide to change the config presets, adjust text here.

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [inline docs/usage/dependency-dashboard.md:74]: Do errorred out branches/updates result in a issue being created, or do they fail silently without the dashboard?

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [inline docs/usage/dependency-dashboard.md:100]: Question about security updates when you require approval for _all_ updates.

**Review by HonkingGoose — COMMENTED** [review]
Commenting the current TODO comments.

File: docs/usage/dependency-dashboard.md
Comment (viceice) [inline docs/usage/dependency-dashboard.md:37]:
``````suggestion
Or set `dependencyDashboard` to `true`:

```
{
  "dependencyDashboard": true
}
``````

I wouldn't show here that dashboard approval will enable the dashboard too.

File: docs/usage/dependency-dashboard.md
Comment (viceice) [reply docs/usage/dependency-dashboard.md:46]: Yeah, `dependencyDashboardApproval=true` automatically sets `dependencyDashboard=true` internally. But i think we won't need to metion it here.

File: docs/usage/dependency-dashboard.md
Comment (viceice) [reply docs/usage/dependency-dashboard.md:100]: I think approval for all means approval for all, even security updates.

I think we shouldn't show how to enable approval for **all**, as we strictly discourage it.

File: docs/usage/dependency-dashboard.md
Comment (viceice) [inline docs/usage/dependency-dashboard.md:130]: We should use a package rule instead the shortcut obejects. They are more clear, as a package rule will override short object settings.

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [reply docs/usage/dependency-dashboard.md:37]: Oh whoops, I got the config mixed up here! 😄 I intended to show the other way to turn on the dashboard! 😄

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [reply docs/usage/dependency-dashboard.md:130]: I copy/pasted this from https://docs.renovatebot.com/configuration-options/#dependencydashboardapproval

> If you want to require approval for major updates, set `dependencyDashboardApproval` to `true` within a `major` object:

If this example is bad, we should fix it over there as well. 😄

File: docs/usage/dependency-dashboard.md
Comment (HonkingGoose) [reply docs/usage/dependency-dashboard.md:100]: > I think approval for all means approval for all, even security updates.

Hmm, at least the behavior is consistent! 😄

File: docs/usage/dependency-dashboard.md
Comment (viceice) [reply docs/usage/dependency-dashboard.md:130]: It's not bad, but generally using package rules is best practise. @rarkins WDYT?

**Comment by HonkingGoose** [comment]
I'll stop pushing new stuff into this branch.

I have a feeling I'll need to change filename/directory structure because of #11202 anyways... Or maybe I even need to target a different branch `docs/getting-started-usecases` so that we can do the rework in one big swoop, and review the docs as a coherent whole.

Can you let me know how you want me to proceed @rarkins?
I'll have time to work on this tomorrow.

**Comment by HonkingGoose** [comment]
@rarkins What do you want me to do with the filename/directory structure of this PR?

I think the new `dependency-dashboard.md` file could go in the sidebar under the `getting-started` fold-out. Or we could have the `dependency-dashboard.md` page as a separate entry.

We could end up with this:

- `getting-started`
  - `dependency-dashboard.md`
  - `installing-onboarding.md`
  - `private-packages.md`
  - `running.md`
  - `use-cases.md`

Or we could put `dependency-dashboard.md` in the `docs/usage` folder on its own.

**Comment by rarkins** [comment]
@HonkingGoose I don't think we should overload Getting Started too much. I think we have some core concepts like dashboard, scheduling, grouping and automerge which could together be in a common folder, such as "Key Concepts"?

**Comment by rarkins** [comment]
Unfortunately I don't have any great way to do that.

Something we definitely want to do differently is move all *building* of this repo's modules docs here, instead of in renovatebot.github.io. Then the latter repo is more just pulling together the content instead of generating it.

The next step might be to document in the other repo how to replace the current submodules reference with a branch you're working on. After that you could verify the docs locally by running the build command and then `cd site && npx http-server`.

**Comment by HonkingGoose** [comment]
> Unfortunately I don't have any great way to do that.

Oh okay... Figured it was worth asking, maybe I missed some way to get a preview with the current setup...

> Something we definitely want to do differently is move all _building_ of this repo's modules docs here, instead of in renovatebot.github.io. Then the latter repo is more just pulling together the content instead of generating it.

That would at least mean that all the docs stuff is in one place? Right now we generate a bunch of docs based on the `.ts` files, so I'm not seeing any of that while I'm working on the docs.

> The next step might be to document in the other repo how to replace the current submodules reference with a branch you're working on. After that you could verify the docs locally by running the build command and then `cd site && npx http-server`.

That sounds really clunky, but I'll take it for now, because its still better than no preview method at all. 😄 
I've never even _used_ Git Submodules, so the explanation should be very beginner friendly! 😄 

I think the Docusaurus 2.0 migration has stalled, or maybe the priorities are elsewhere for now? I haven't seen any work on that in a while now.

---

@rarkins Could you maybe comment on the PR comments how you want me to resolve those? I don't know enough yet to proceed with the document. I think you're waiting on me to release the Dependency Dashboard page for your `v26` release?

**Comment by HonkingGoose** [comment]
> > @rarkins Could you maybe comment on the PR comments how you want me to resolve those? I don't know enough yet to proceed with the document. I think you're waiting on me to release the Dependency Dashboard page for your `v26` release?
> 
> Do you mean in this PR?

Yup! 😉

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:10]:
```suggestion
When the Dependency Dashboard is enabled, Renovate will create a new issue in the repository.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:13]:
```suggestion
Having the Dependency Dashboard also enables the concept of an "approval" workflow for new upgrades, either for selected dependencies (recommended) or even for all.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:17]:
```suggestion
The Dependency Dashboard requires that the host platforms supports the concept of issues with dynamic markdown checkboxes.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:23]:
```suggestion
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:47]:
```suggestion
```

Turning on any approvals implicitly enables the dashboard, so the documentation is correct

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:52]:
```suggestion
To disable the Dependency Dashboard, add the preset `:disableDependencyDashboard` or set `dependencyDashboard` to `false`.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:51]:
```suggestion
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:59]:
```suggestion
```json
{
  "extends": ["config:base", ":disableDependencyDashboard"]
}
```

or

```json
{
  "extends": ["config:base"],
  "dependencyDashboard": false
}
```
```

Comment (rarkins) [reply docs/usage/key-concepts/dependency-dashboard.md:59]: not sure the preview is correct here? I think the code is

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:61]:
```suggestion
## Use cases
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:63]:
```suggestion
This section explains some common use cases where having the Dependency Dashboard can help.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:69]:
```suggestion
If you close an update PR from Renovate without merging, the Dashboard will list this update in the Closed/Ignored section.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:75]:
```suggestion
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:93]:
```suggestion
You'll probably forgot to check often enough, and out of sight means out of mind!
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:101]:
```suggestion
However, vulnerability remediation PRs will still get created immediately without requiring approval.

```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:103]:
```suggestion
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:104]:
```suggestion
To require manual approval for _all updates_, add the `:dependencyDashboardApproval` presets to the `extends` array in your Renovate configuration file:
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:110]:
```suggestion
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:118]:
```suggestion
Major updates often contain breaking changes which require manual changes in your code before they can be merged.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:119]:
```suggestion
So maybe you only want to get major updates when you have sufficient time to check them carefully.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (rarkins) [inline docs/usage/key-concepts/dependency-dashboard.md:147]:
```suggestion
      "matchPackagePatterns": ["^@somescope"],
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (HonkingGoose) [reply docs/usage/key-concepts/dependency-dashboard.md:17]:
```suggestion
The Dependency Dashboard requires that the host platforms supports the concept of issues with dynamic Markdown checkboxes.
```

File: docs/usage/key-concepts/dependency-dashboard.md
Comment (HonkingGoose) [reply docs/usage/key-concepts/dependency-dashboard.md:59]: I'll check after applying the suggestion! 😄

**Comment by HonkingGoose** [comment]
I'll just post this again, as it will get missed behind the wall of text we have now... 😉 

> Do we want to have screenshots of the dashboard?
> 
> **Benefits:**
> 
> - Pictures help to liven up the page so it's not just a wall of text
> - Pictures are a good way to show why you want to have the dashboard
> - Helps market the feature to new prospecting users who are reading the docs to find out if Renovate is a good fit for them
> 
> **Drawbacks:**
> 
> - Pictures will need to be updated to match current design (example when we implement issue `#10924`)
> - Increases size of repository a bit
> - Requires maintainer time to update pictures

File: docs/usage/key-concepts/dashboard.md
Comment (HonkingGoose) [inline docs/usage/key-concepts/dashboard.md:1]: Do we want images/screenshots in this document? See the bottom post in the PR for my comments/thoughts.

Resolve this PR comment once we have reached a decision.

**Comment by rarkins** [comment]
Screenshots would be great, but done is sometimes better than perfect :)

As in, we could defer them to a future PR instead of holding this up

File: docs/usage/key-concepts/dashboard.md
Comment (rarkins) [reply docs/usage/key-concepts/dashboard.md:1]: Let's merge without them, but definitely it would be better to have them added

**Comment by HonkingGoose** [comment]
> Screenshots would be great, but done is sometimes better than perfect :)
> 
> As in, we could defer them to a future PR instead of holding this up

Let's go with "defer to future PR" then and get this important work merged! 😄 

I'll make a new issue to track this proposal. 😄

**Review by rarkins — APPROVED** [review]

**Pull request merged**
